pallets/gear/src/lib.rs

```diff
@@ -101,6 +102,14 @@ impl DebugInfo for () {
     }
 }
 
+#[derive(Clone, Debug, Decode, Encode, PartialEq, Eq, scale_info::TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+pub struct GasInfo {
+    pub spent: u64,
+    pub to_send: u64,
+    pub burnt: u64,
+}
```

Comment:
```suggestion
#[derive(Clone, Debug, Decode, Encode, PartialEq, Eq, scale_info::TypeInfo)]
#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
pub struct GasInfo {
    pub spent: u64,
    pub to_send: u64,
    pub burned: u64,
}

impl GasInfo {
    pub fn balance_diff(&self) -> u64 {
        self.to_send.saturating_add(self.burned)
    }
}
```

Reply:
I also missed the fact that exactly gas can't be sent to anybody - we can paste here gas_reserved (which will be used for charging rent for anything) and VALUE to be sent, which is u128, so maybe need to consider about renaming GasInfo into some EconomicInfo/BalanceInfo

Reply:
`GasInfo` is lightweight enough for me to be also `Copy`

Reply:
Some docs are also need